sail/action.yml

```diff
@@ -0,0 +1,88 @@
+    required: true
+  initialize_port_resources:
+    description: 'Should ocean try to create the default blueprints, pages & integration config for the integration'
+    required: false
+    default: 'true'
+  config:
+    description: 'The configuration for the integration'
+    required: false
+
+  # Misc inputs
+  platform:
+    description: 'The platform to run the integration on'
+    required: false
+    default: 'linux/amd64'
+  image:
+    description: 'The image to run the integration from'
+    required: false
+  version:
+    description: 'The version of the integration to run'
+    required: false
+    default: 'latest'
+outputs:
+  exit_code:
+    description: 'The exit code of the Ocean Sail command'
+    value: ${{ steps.ocean-sail.outputs.exit_code }}
+runs:
+  using: "composite"
+  steps:
+    - name: ✅ Validate inputs
+      shell: bash
+      run: |
+        [[ "${{ inputs.type }}" ]] || { echo "type input is empty" ; exit 1; }
+        [[ "${{ inputs.port_client_id }}" ]] || { echo "inputs.port_client_id input is empty" ; exit 1; }
+        [[ "${{ inputs.port_client_secret }}" ]] || { echo "inputs.port_client_secret input is empty" ; exit 1; }
+        [[ "${{ inputs.initialize_port_resources }}" ]] || { echo "inputs.initialize_port_resources input is empty" ; exit 1; }
+        [[ "${{ inputs.event_listener_type }}" ]] || { echo "inputs.event_listener_type input is empty" ; exit 1; }
+        [[ "${{ inputs.platform }}" ]] || { echo "inputs.platform input is empty" ; exit 1; }
+        [[ "${{ inputs.version }}" ]] || { echo "inputs.version input is empty" ; exit 1; }
+
+    - name: 🌊 Run Ocean Sail dockerized
+      shell: bash
+      run: |
+        image_name="ghcr.io/port-labs/port-ocean-${{ inputs.type }}:${{ inputs.version }}"
+        if [[ "${{ inputs.image }}" ]]; then
+          image_name="${{ inputs.image }}:${{ inputs.version }}"
+        fi
+        
+        echo "OCEAN__PORT__CLIENT_ID=${{ inputs.port_client_id }}" >> .env
+        echo "OCEAN__PORT__CLIENT_SECRET=${{ inputs.port_client_secret }}" >> .env
+        
+        echo "OCEAN__EVENT_LISTENER={\"type\":\"ONCE\"}" >> .env
+        echo "OCEAN__INITIALIZE_PORT_RESOURCES=${{ inputs.initialize_port_resources }}" >> .env
+        
+        if [[ "${{ inputs.identifier }}" ]]; then
+          echo "OCEAN__INTEGRATION__IDENTIFIER=${{ inputs.identifier }}" >> .env
+        fi
+        
+        if [[ "${{ inputs.config }}" ]]; then
+          for key in $(echo "${{ inputs.config }}" | yq eval "keys | .[]"); do
+            value=$(echo "${{ inputs.config }}" | yq eval ".$key")
+            upper_key=$(echo $key | tr '[:lower:]' '[:upper:]')
+            echo "OCEAN__INTEGRATION__CONFIG__${upper_key}=${value}" >> .env
+          done
+        fi
+        
+        env >> .env
+        
+        docker run -i --rm --platform=${{ inputs.platform }} --env-file .env $image_name
+        
+        echo "exit_code=$?" >> $GITHUB_ENV
```

Comment:
Will this code section be reached even if the docker run returns a non-zero exit code?

And if it might not, then the next step for cleanup needs to always happen to make sure that the cleanup always happens

Reply:
added if: always
and yes it will
took it from the once docs that we have